clientd3d/audio_openal.c

```diff
@@ -0,0 +1,843 @@
+{
+   if (g_initialized)
+      return true;
+
+   // Open default audio device
+   g_device = alcOpenDevice(NULL);
+   if (!g_device)
+   {
+      debug(("AudioInit: Failed to open audio device\n"));
+      return false;
+   }
+
+   // Create audio context
+   g_context = alcCreateContext(g_device, NULL);
+   if (!g_context)
+   {
+      debug(("AudioInit: Failed to create audio context\n"));
+      alcCloseDevice(g_device);
+      g_device = NULL;
+      return false;
+   }
+
+   // Make context current
+   if (!alcMakeContextCurrent(g_context))
+   {
+      debug(("AudioInit: Failed to make context current\n"));
+      alcDestroyContext(g_context);
+      alcCloseDevice(g_device);
+      g_context = NULL;
+      g_device = NULL;
+      return false;
+   }
+
+   // Generate audio sources
+   alGenSources(MAX_AUDIO_SOURCES, g_sources);
+   if (alGetError() != AL_NO_ERROR)
+   {
+      debug(("AudioInit: Failed to generate audio sources\n"));
+      alcMakeContextCurrent(NULL);
+      alcDestroyContext(g_context);
+      alcCloseDevice(g_device);
+      g_context = NULL;
+      g_device = NULL;
+      return false;
+   }
+   g_numSources = MAX_AUDIO_SOURCES;
+
+   // Create dedicated music source
+   alGenSources(1, &g_musicSource);
+   if (alGetError() != AL_NO_ERROR)
+   {
+      debug(("AudioInit: Failed to generate music source\n"));
+   }
+
+   // Set default listener properties
+   alListener3f(AL_POSITION, 0.0f, 0.0f, 0.0f);
+   alListener3f(AL_VELOCITY, 0.0f, 0.0f, 0.0f);
+   ALfloat listenerOri[] = { 0.0f, 0.0f, -1.0f, 0.0f, 1.0f, 0.0f }; // forward, up
+   alListenerfv(AL_ORIENTATION, listenerOri);
+
+   // Use linear distance model for predictable 3D audio falloff
+   alDistanceModel(AL_LINEAR_DISTANCE_CLAMPED);
+
+   g_initialized = true;
+   debug(("AudioInit: OpenAL initialized successfully\n"));
+   
+   return true;
+}
+
+/*
+ * AudioShutdown: Clean up audio system
+ */
+void AudioShutdown(void)
+{
+   if (!g_initialized)
+      return;
+
+   // Stop all sounds
+   SoundStopAll();
+   MusicStop();
+
+   // Delete music source
+   if (g_musicSource)
+   {
+      alDeleteSources(1, &g_musicSource);
+      g_musicSource = 0;
+   }
+
+   // Delete sound sources
+   if (g_numSources > 0)
+   {
+      alDeleteSources(g_numSources, g_sources);
+      g_numSources = 0;
+   }
+
+   // Delete all cached buffers
+   for (auto& node : g_cacheList)
+   {
+      alDeleteBuffers(1, &node.buffer);
+   }
+   g_cacheList.clear();
+   g_cacheMap.clear();
+
+   // Destroy context
+   if (g_context)
+   {
+      alcMakeContextCurrent(NULL);
+      alcDestroyContext(g_context);
+      g_context = NULL;
+   }
+
+   // Close device
+   if (g_device)
+   {
+      alcCloseDevice(g_device);
+      g_device = NULL;
+   }
+
+   g_initialized = false;
+   debug(("AudioShutdown: OpenAL shut down\n"));
+}
+
+/*
+ * IsBufferInUse: Returns true if the buffer is currently being played by any source.
+ * Used to prevent evicting buffers that are actively playing.
```

Comment:
Document what the function does, not how you expect it to be called.  You don't know whether it's going to be used to prevent evicting buffers in the future.